File: easy-level/20. valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var parentheses = {
    "(": ")",
    "{": "}",
    "[": "]"
};

var isValid = function (s) {
    var stack = [];

    for (var i = 0; i < s.length; i++) {
        if (s[i] === parentheses[stack[stack.length - 1]]) stack.pop();
        else stack.push(s[i]);
    }

    return stack.length === 0;
};

File: easy-level/20. valid-parentheses.py
parentheses = {
    "(": ")",
    "{": "}",
    "[": "]"
}

class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        for i in s:
            if len(stack) > 0 and stack[-1] in parentheses.keys() and i == parentheses[stack[-1]]:
                stack.pop()
            else:
                stack.append(i)

        return len(stack) == 0